### ISSUE TITLE
Better Lua interface docs with LDoc

### ISSUE THREAD

**Issue by DSpeichert** [otland/forgottenserver#2384]
Let's use [ldoc](https://github.com/stevedonovan/ldoc) for documenting all Lua interfaces.

See example: https://stevedonovan.github.io/Penlight/api/index.html

**Comment by pedbernardo** [comment]
Any progress in this regard?

I'm not used to Lua, [LDoc](https://github.com/stevedonovan/LDoc) still make sense?
Personally, I'm trying to learn TFS + Lua but due to lack of good documentation and help from community on Discord it's being hard understand function usage, params, returns, scope...

Seems crazy so many time of developing without that, or I'm losing something?

**Comment by soul4soul** [comment]
> 
> 
> Any progress in this regard?
> 
> I'm not used to Lua, [LDoc](https://github.com/stevedonovan/LDoc) still make sense?
> Personally, I'm trying to learn TFS + Lua but due to lack of good documentation and help from community on Discord it's being hard understand function usage, params, returns, scope...
> 
> Seems crazy so many time of developing without that, or I'm losing something?

Have you see the [wiki](https://github.com/otland/forgottenserver/wiki) for this repo? At this time its the only written lua documentation that I'm aware of for TFS.

**Comment by pedbernardo** [comment]
@soul4soul 

Yep. I saw... But still lacks many information. Hasn't all functions documented and some are incomplete.
I tried LDoc to integrate with VSCode, like JSDoc does. Sadly, has no native support or a plugin.

I find this too: https://stigmax.gitbook.io/tfs-guide
_Incomplete too..._
